lib/orderbook/OrderBook.ts

```diff
@@ -80,7 +80,18 @@ class OrderBook {
     const { matches, remainingOrder } = matchingResult;
 
     if (matches.length > 0) {
-      matches.forEach(this.handleMatch);
+      for (const [index, match] of matches.entries()) {
+        const { maker, taker } = match;
+        this.handleMatch({ maker, taker });
+
+        // Maker is fully matched if it is not the last
+        if ((matches.length - index) > 1) {
+          await this.repository.updateOrder(maker, 0);
+        } else {
+          const initialQuantity = await this.repository.getOrderQuantity(maker);
+          await this.repository.updateOrder(maker, initialQuantity - maker.quantity);
```

Comment:
These two operations can be reduced to one, atomic. `sequelize` even expose an interface for it, so you won't have to do SQL for that. Here's an example: https://github.com/sequelize/sequelize/issues/7268